Kinvey.Tests/DataStoreTests/DataStoreNetworkIntegrationTests.cs

```diff
@@ -5089,204 +5234,247 @@ public async Task TestNetworkStoreGetSumAsync()
+            kinveyClient = BuildClient("5");
+
             if (MockData)
             {
-                MockResponses(1);
+                MockResponses(4);
             }
 
-            //Arrange
             await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK, kinveyClient);
+            // Arrange
+            var newItem = new ToDo
+            {
+                Name = "Next Task",
+                Details = "A test",
+                DueDate = "2016-04-19T20:02:17.635Z"
+            };
+            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
 
             // Act
-            var exception = Assert.ThrowsException<KinveyException>(delegate
-            {
-                todoStore.GetSyncCount();
-            });
+            var savedToDo = await todoStore.SaveAsync(newItem);
+
+            var existingToDo = await todoStore.FindByIDAsync(savedToDo.ID);
+
+            // Teardown
+            await todoStore.RemoveAsync(savedToDo.ID);
 
             // Assert
-            Assert.AreEqual(typeof(KinveyException), exception.GetType());
-            var kinveyException = exception as KinveyException;
-            Assert.AreEqual(EnumErrorCategory.ERROR_DATASTORE_NETWORK, kinveyException.ErrorCategory);
-            Assert.AreEqual(EnumErrorCode.ERROR_DATASTORE_INVALID_SYNC_COUNT_OPERATION, kinveyException.ErrorCode);
+            Assert.IsNotNull(existingToDo);
+            Assert.AreEqual(newItem.Name, existingToDo.Name);
+            Assert.AreEqual(newItem.Details, existingToDo.Details);
+            Assert.AreEqual(newItem.DueDate, existingToDo.DueDate);
         }
 
         [TestMethod]
-        public async Task TestClearCacheAsync()
+        public async Task TestSaveUpdateApiVersion5Async()
         {
             // Setup
+            kinveyClient = BuildClient("5");
+
             if (MockData)
             {
-                MockResponses(1);
+                MockResponses(4);
             }
 
-            //Arrange
             await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK, kinveyClient);
+            // Arrange
+            var newItem = new ToDo
+            {
+                ID = Guid.NewGuid().ToString(),
+                Name = "Next Task",
+                Details = "A test",
+                DueDate = "2016-04-19T20:02:17.635Z"
+            };
+            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
 
             // Act
-            var exception = Assert.ThrowsException<KinveyException>(delegate
-            {
-                todoStore.ClearCache();
-            });
+            var savedToDo = await todoStore.SaveAsync(newItem);
+
+            var existingToDo = await todoStore.FindByIDAsync(savedToDo.ID);
```

Comment:
I think it would be better to use just Find and assert the number of items - if for some reason two item is recorded multiple times

Reply:
Ok.